### PR TITLE
issue 900 - fixed bug with allowing inactive users with a valid jwt t…

### PR DESCRIPTION
**Pull request by fourgates**
https://github.com/Requarks/wiki/issues/900 - fixed bug with allowing inactive users with a valid jwt to continue viewing non-guest content. It has been documented in the wikijs issue. 

This is my first Wiki.js PR and I have never used this stack before, so let me know if I am not following conventions.  I used `authentication\local\authentication.js` as a reference on how to load a user.

### CODE REVIEW

**Comment by fourgates** [comment]
https://github.com/Requarks/wiki/issues/900

File: server/core/auth.js [hunk 1]
```diff
@@ -125,7 +125,16 @@ module.exports = {
           return next()
         }
       }
-
+      // JWT is valid but user is NOT active
+      if (user) {
+        const dbUser = await WIKI.models.users.query().findOne({
+          email: user.email,
+          providerKey: 'local'
```
Comment (NGPixel) [inline server/core/auth.js]: Why is local hardcoded? Should be using whatever provider the user currently has.

Comment (fourgates) [reply server/core/auth.js]: I updated the PR to use the provider on the user.

**Comment by NGPixel** [comment]
Just noticed that this queries the DB on every request, which is a big no go. The check should ideally be done in the `refreshToken()` method of the users model. I'll make the necessary changes (unless you want to it).

**Comment by fourgates** [comment]
@NGPixel that looks like it would be an easy change. Besides the obvious of not hitting the db on every request, why do you think thats the best place? When does it get called?

**Comment by fourgates** [comment]
@NGPixel did you address this issue?

**Comment by regevbr** [comment]
@NGPixel you can close this pr. The fix suggested here is indeed not good and validation on token refresh is already implemented

**Pull request closed without merging**